Makefile

```diff
@@ -57,7 +57,7 @@ install: clean ## Install python package.
 	python setup.py install
 
 .PHONY: lint
-lint: test-black test-codespell test-flake8 test-isort test-mypy test-pydocstyle test-pylint test-pyright ## Run all linting tests.
+lint: test-black test-codespell test-flake8 test-isort test-mypy test-pydocstyle test-pyright test-pylint ## Run all linting tests.
```

Comment:
is there a reason for changing this?

Reply:
Possibly when @cjdcordeiro was working on this the `pylint` step was broken (I think I fixed it in the meantime)

Reply:
yes. the reason is to be consistent with the CI.

the `linters` job (https://github.com/canonical/rockcraft/blob/main/.github/workflows/tests.yaml#L10) isn't (yet) calling for pylint. Locally, I like to run `make tests` before pushing new changes to ensure the CI will pass. With the previous order, if pylint complains, `make tests` will fail before getting to `pyright`, so you won't get an accurate prediction of the CI (unless you then manually run the `pyright` by hand. 

Another alternative would be to have a linting pipeline that doesn't fail-fast, always going through all tests, so that you can run all linters in one go, and get all issues that need fixing